cpp/validate_harvested_records.cc

```diff
@@ -322,126 +244,122 @@ void SendEmail(const std::string &email_address, const std::string &message_subj
 }
 
 
-void UpdateDB(DbConnection * const db_connection, const std::string &zeder_id, const std::string &zeder_instance,
-              const std::string &field_name, const std::string &field_presence_str)
-{
-    FieldPresence field_presence;
-    if (not StringToFieldPresence(field_presence_str, &field_presence))
-        LOG_ERROR("\"" + field_presence_str + "\" is not a valid field_presence!");
-    if (field_name.length() != MARC::Record::TAG_LENGTH)
-        LOG_ERROR("\"" + field_name + "\" is not a valid field name!");
-
-    db_connection->queryOrDie("UPDATE metadata_presence_tracer SET field_presence='" + field_presence_str + "' WHERE zeder_journal_id="
-                              + "(SELECT id FROM zeder_journals WHERE zeder_id=" + db_connection->escapeAndQuoteString(zeder_id) + " "
-                              + "AND zeder_instance=" + db_connection->escapeAndQuoteString(zeder_instance) + ") "
-                              + "AND field_name='" + field_name + "'");
-    if (db_connection->getNoOfAffectedRows() == 0)
-        LOG_ERROR("can't update non-existent database entry: " + zeder_id + "(" + zeder_instance + ")"
-                  + ", field_name: \"" + field_name + "\"");
+void CheckGenericRequirements(const MARC::Record &record, std::string * const reasons_for_being_invalid) {
+    if (not record.hasTag("001"))
+        reasons_for_being_invalid->append("control field 001 is missing\n");
+
+    if (not record.hasTag("008"))
+        reasons_for_being_invalid->append("control field 008 is missing\n");
+
+    const auto _245_field(record.findTag("245"));
+    if (_245_field == record.end())
+        reasons_for_being_invalid->append("field 245 is missing\n");
+    else if (_245_field->getFirstSubfieldWithCode('a').empty())
+        reasons_for_being_invalid->append("subfield 245$a is missing\n");
+
+    if (not record.hasTag("100") and not record.hasTag("700"))
```

Comment:
I am not sure if it is really mandatory to have at least 1 author

Reply:
According to my conversation w/ C.W. last week that is a hard requirement.

Reply:
Then maybe we should put a documentation into the zotero_cgi webinterface (where the rules are listed) and tell the user which additional hardcoded checks are performed and where there is no need to add explicit rules e.g. for 100 and 700.

Reply:
Also we should put a comment somewhere here that if somebody makes any changes here the frontend documentation should be changed as well